statsmodels/stats/contingency_tables.py

```diff
@@ -0,0 +1,1312 @@
+
+
+
+def _make_df_square(table):
+    """
+    Reindex a pandas DataFrame so that it becomes square, meaning that
+    the row and column indices contain the same values, in the same
+    order.  The row and column index are extended to achieve this.
+    """
+
+    if not isinstance(table, pd.DataFrame):
+        return table
+
+    # If the table is not square, make it square
+    if table.shape[0] != table.shape[1]:
+        ix = list(set(table.index) | set(table.columns))
+        table = table.reindex(ix, axis=0)
+        table = table.reindex(ix, axis=1)
+
+    # Ensures that the rows and columns are in the same order.
+    table = table.reindex(table.columns)
+
+    return table
+
+
+class _Bunch(object):
+    pass
+
+
+class Table(object):
+    """
+    Analyses that can be performed on a two-way contingency table.
+
+    Parameters
+    ----------
+    table : array-like
+        A contingency table.
+    shift_zeros : boolean
+        If True and any cell count is zero, add 0.5 to all values
+        in the table.
+
+    Attributes
+    ----------
+    table_orig : array-like
+        The original table is cached as `table_orig`.
+    nominal_association : bunch
+        Assessment of independence between rows and columns using chi^2
+        testing.  The rows and columns are treated as nominal (unordered)
+        categorical ariables.  The returned bunch includes the following
+        attributes:
+            statistic : float
+                The chi^2 test statistic.
+            df : integer
+                The degrees of freedom of the reference distribution
+            pvalue : float
+                The p-value for the test.
+    marginal_probabilities : tuple of two arrays
+        The estimated row and column marginal distributions.
+    independence_probabilities : array-like
+        Estimated cell probabilities under independence.
+    fittedvalues : array-like
+        Fitted values under independence.
+    pearson_resids : array-like
+        The Pearson residuals.
+    standardized_resids : array-like
+        Residuals with approximately unit variance.
+    chi2_contribs : array-like
+        The contribution of each cell to the chi^2 statistic.
+    local_logodds_ratios : array-like
+        The local log odds ratios are calculated for each 2x2 subtable
+        formed from adjacent rows and columns.
+    local_oddsratios : array-like
+        The local odds ratios are calculated from each 2x2 subtable
+        formed from adjacent rows and columns.
+    cumulative_log_oddsratios : array-like
+        The cumulative log odds ratio at a given pair of thresholds is
+        calculated by reducing the table to a 2x2 table based on
+        dichotomizing the rows and columns at the given thresholds.
+        The table of cumulative log odds ratios presents all possible
+        cumulative log odds ratios that can be formed from a given
+        table.
+    cumulative_oddsratios : array-like
```

Comment:
`array_like` is not well defined for Returns or attributes, `ndarray` or `DataFrame`?